### ISSUE TITLE
getEndpointMetadata in v1.2 not compatible with DigitalOcean spaces api

### ISSUE THREAD

**Issue by mitalcoi** [async-aws/aws#688]
because of hardcoding api endpoint in getEndpointMetadata on v1.2 AsyncAws\S3\S3Client now is not compatible with DigitalOcean or any other s3-compatible vendors. I have to downgrade on v1.1.0 to continue work with DO Spaces

**Comment by jderusse** [comment]
Thanks for reporting this issue.

The endpoints provided by `getEndpointMetadata` should be used when you defined your own endpoint.
Could you please share your configuration?

Side note: did you enabled the option `pathStyleEndpoint=true` (Digital Ocean probably did not implement AWS virtual host) https://async-aws.com/clients/s3.html#virtual-hosted-style-requests

**Comment by mitalcoi** [comment]
My config:
<img width="679" alt="Снимок экрана 2020-06-12 в 14 52 04" src="https://user-images.githubusercontent.com/3224226/84500003-5948e500-acbc-11ea-9125-a9a1eec0fccd.png">


I not checked with pathStyleEndpoint=true, I'm afraid DO not support it

**Comment by jderusse** [comment]
That's probably your issue. Try to set the option to true (default is false).

AWS use HostStyle but is also compatible with PathStyle. DO is only compatible with path style. that why you have to set the option to true to force AsyncAws using PathStyle.

**Comment by Nyholm** [comment]
@mitalcoi did pathStyleEndpoint work for you?

**Comment by mitalcoi** [comment]
no, the same problem (
```
S3_REGION=fra1
S3_TOKEN_KEY=***
S3_TOKEN_SECRET=***
S3_MEDIA_BUCKET=***
S3_ENDPOINT=https://fra1.digitaloceanspaces.com
```

<img width="698" alt="Снимок экрана 2020-06-15 в 15 46 52" src="https://user-images.githubusercontent.com/3224226/84659020-716c6e80-af1f-11ea-947b-9464be87f7be.png">

**Comment by Nyholm** [comment]
I've started on a small test in https://github.com/async-aws/aws/pull/697

Could you help me finish it to show what is wrong?

**Comment by mitalcoi** [comment]
yes, I will pay attention

**Comment by Nyholm** [comment]
Closed by #697
